Models/ProductsListPart.cs

```diff
@@ -1,35 +1,90 @@
 ﻿using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using Orchard.ContentManagement;
+using Orchard.ContentManagement.FieldStorage.InfosetStorage;
 using Orchard.ContentManagement.Utilities;
 using Orchard.Environment.Extensions;
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Xml.Linq;
 
 namespace Nwazet.Commerce.Models {
     [OrchardFeature("Nwazet.PersistentCart")]
     public class ProductsListPart : ContentPart {
+
+        private const string partName = "ProductsListPart";
+
         //This part goes in the infoset
 
         private IEnumerable<ShoppingCartItem> _items {
             get {
-                var fromInfoset = Retrieve<string>("SerializedItems");
-                if (string.IsNullOrWhiteSpace(fromInfoset)) {
-                    return null;
-                }
-                var jsonInfoset = JObject.Parse("{'List':" + fromInfoset + "}");
-                return jsonInfoset["List"].Children()
-                    .Select(jt => {
-                        return new ShoppingCartItem(
-                            productId: int.Parse(jt["ProductId"].ToString()),
-                            quantity: int.Parse(jt["Quantity"].ToString()),
-                            attributeIdsToValues: JsonConvert.DeserializeObject<Dictionary<int, ProductAttributeValueExtended>>(jt["AttributeIdsToValues"].ToString())
-                            );
-                    })
-                    .ToList();
+                return RetrieveItems().ToList();
             }
             set {
-                Store("SerializedItems", JsonConvert.SerializeObject(value));
+                Store(value);
+            }
+        }
+
+        private IEnumerable<ShoppingCartItem> RetrieveItems() {
+            var partElement = GetInfosetElement();
+            return partElement.Elements()
+                .Where(el => el.Name.LocalName.StartsWith("ShoppingCartItem_")) //XElements corresponding to items
+                .Select(el =>  //parse the XElement to an item
+                    new ShoppingCartItem(
+                        productId: int.Parse(el.Attribute("ProductId").Value),
+                        quantity: int.Parse(el.Attribute("Quantity").Value),
+                        attributeIdsToValues: el.Elements()
+                            .Where(subel => subel.Name.LocalName.StartsWith("AttributeIdsToValues_")) //XElements corresponding to dictionary entries
```

Comment:
Why do these need to have different tag names? Looks like child elements would work better and be easier.